### PR TITLE
check for error type only when expecting error.

### PR DESCRIPTION
**Pull request by mchoube**
incorporated code review comment.

### CODE REVIEW

**Comment by petertseng** [comment]
Two suggestions on how to achieve "Only check expected value if there is no error":

Binary uses a normal `if/else` https://github.com/exercism/xgo/blob/master/exercises/binary/binary_test.go#L53

Trinary uses a `switch/case`: https://github.com/exercism/xgo/blob/master/exercises/trinary/trinary_test.go#L40

**Comment by mchoube** [comment]
what about
if !tt.expectErr && err != nil {

should error be checked when not expected?

**Comment by petertseng** [comment]
> `if !tt.expectErr && err != nil `
> should error be checked when not expected?

That line should stay because if the student submits a solution that gives a non-nil error when error should be nil, the test should fail.

So if the test case does not expect an error, then:

* The student's function should not say there was an error
* We should also check if the non-error return value (`actualVerse`) is correct.

If the test case does expect an error:

* We should check that the student's function also says there is an error (and it is of the error interface)
* We do *not* check the non-error return value (`actualVerse`)

File: exercises/beer-song/beer_test.go [hunk 1]
```diff
@@ -57,18 +57,23 @@ var verseTestCases = []struct {
...
-		}
-
-		// if we expect an error and there isn't one
-		if tt.expectErr && err == nil {
-			t.Errorf("Verse(%d): expected an error, but error is nil", tt.verse)
-		}
-		// if we don't expect an error and there is one
-		if !tt.expectErr && err != nil {
-			t.Errorf("Verse(%d): expected no error, but error is: %s", tt.verse, err)
+		if tt.expectErr {
```
Comment (petertseng) [inline exercises/beer-song/beer_test.go]: Thanks! Looks like what you did here is good!

File: exercises/beer-song/beer_test.go [hunk 2]
```diff
@@ -91,10 +96,14 @@ func TestSeveralVerses(t *testing.T) {
 
 	for _, tt := range versesTestCases {
 		actualVerse, err := Verses(tt.upperBound, tt.lowerBound)
-		var _ error = err
 		if actualVerse != tt.expectedVerse {
```
Comment (petertseng) [inline exercises/beer-song/beer_test.go]: could you do the same for this `if actualVerse != tt.expectedVerse` ?

Comment (mchoube) [inline exercises/beer-song/beer_test.go]: oops missed that. will update and send review.

Comment (mchoube) [inline exercises/beer-song/beer_test.go]: fixed please review.

**Comment by petertseng** [comment]
Thanks! looking good! I will be squashing these three commits into one.

**Pull request merged**